<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>569</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>elfparser-ng</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/locked</normaloff>:/locked</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QFrame" name="leftFrame">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="scoreBox">
         <property name="title">
          <string>Score</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLCDNumber" name="scoreDisplay">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>60</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="controlsBox">
         <property name="title">
          <string>Controls</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="1">
           <widget class="QPushButton" name="closeButton">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>Close</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="openButton">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Utils</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="1">
           <widget class="QPushButton" name="hexButton">
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>HexEditor</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="resetButton">
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="rpasserButton">
            <property name="text">
             <string>Rpasser</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="leftFrameSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="mainTabs">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="currentIndex">
       <number>6</number>
      </property>
      <widget class="QWidget" name="overviewTab">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <attribute name="title">
        <string>Overview</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QTableWidget" name="overviewTable">
          <property name="contextMenuPolicy">
           <enum>Qt::ActionsContextMenu</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="dragDropOverwriteMode">
           <bool>false</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="columnCount">
           <number>1</number>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <row>
           <property name="text">
            <string>Filename</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Size</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>md5</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>sha1</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>sha256</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Family</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string/>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="headerTab">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <attribute name="title">
        <string>ELF Header</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QTableWidget" name="headerTable">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <row>
           <property name="text">
            <string>Magic</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Class</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Encoding</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>ELF Version</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>OS ABI</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>ABI Version</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Type</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Machine</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Version</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Entry Point</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>PH Offset</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>SH Offset</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Flags</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>Header Size</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>PH Entry Size</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>PH Entries</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>SH Entry Size</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>SH Entries</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>String Index</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string/>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="sectionsTab">
       <attribute name="title">
        <string>SHeaders</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="sectionsGrou">
          <property name="title">
           <string>Section Headers</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QTableWidget" name="sectionsTable">
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="showDropIndicator" stdset="0">
              <bool>false</bool>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
             <attribute name="horizontalHeaderCascadingSectionResizes">
              <bool>false</bool>
             </attribute>
             <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
              <bool>true</bool>
             </attribute>
             <attribute name="horizontalHeaderStretchLastSection">
              <bool>false</bool>
             </attribute>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string>Index</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Name</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Type</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Flags</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Virtual Address</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Offset</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Size</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Link</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="sectionDetails">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>Details</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPlainTextEdit" name="sectionInfo">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="programsTab">
       <attribute name="title">
        <string>PHeaders</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QGroupBox" name="programsGroup">
          <property name="title">
           <string>Program Headers</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QTableWidget" name="programsTable">
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string>Type</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Offset</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Virtual Address</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Physical Address</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>File Size</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Memory Size</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Flags</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="detailsGroup">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Details</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPlainTextEdit" name="programsInfo">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="plainText">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="symbolsTab">
       <attribute name="title">
        <string>Symbols</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QTableWidget" name="symbolsTable">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Binding</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Value</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="capabilitiesTab">
       <attribute name="title">
        <string>Capabilities</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QTreeWidget" name="capabilitiesTree">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="uniformRowHeights">
           <bool>false</bool>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Category</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Details</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="scoringTab">
       <attribute name="title">
        <string>Scoring</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QTableWidget" name="scoringTable">
          <property name="maximumSize">
           <size>
            <width>10000</width>
            <height>10000</height>
           </size>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="dragDropOverwriteMode">
           <bool>false</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="sortingEnabled">
           <bool>false</bool>
          </property>
          <property name="columnCount">
           <number>2</number>
          </property>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Score</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Reason</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>31</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuabout">
    <property name="title">
     <string>help</string>
    </property>
    <addaction name="aboutButton"/>
   </widget>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="openButton2"/>
   </widget>
   <addaction name="menufile"/>
   <addaction name="menuabout"/>
  </widget>
  <action name="aboutButton">
   <property name="text">
    <string>about</string>
   </property>
  </action>
  <action name="openButton2">
   <property name="text">
    <string>open</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>282</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rpasserButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>rparser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sectionsTable</sender>
   <signal>currentItemChanged(QTableWidgetItem*,QTableWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>sectionSelected(QTableWidgetItem*,QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>551</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>programsTable</sender>
   <signal>currentItemChanged(QTableWidgetItem*,QTableWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>programSelected(QTableWidgetItem*,QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFile()</slot>
  <slot>sectionSelected(QTableWidgetItem*,QTableWidgetItem*)</slot>
  <slot>programSelected(QTableWidgetItem*,QTableWidgetItem*)</slot>
  <slot>reset()</slot>
  <slot>closeAbout()</slot>
  <slot>rparser()</slot>
 </slots>
</ui>
